drivers/can/can_common.c

```diff
@@ -35,6 +35,21 @@ int z_impl_can_send(const struct device *dev, const struct can_frame *frame,
 		    void *user_data)
 {
 	const struct can_driver_api *api = (const struct can_driver_api *)dev->api;
+	uint32_t id_mask;
+
+	if ((frame->flags & CAN_FRAME_IDE) != 0U) {
+		id_mask = CAN_EXT_ID_MASK;
+	} else {
+		id_mask = CAN_STD_ID_MASK;
+	}
+
+	if ((frame->id & ~(id_mask)) != 0U) {
```

Comment:
I am considering changing these to use the `CHECKIF()` macro instead, allowing users to turn off these runtime checks if desired. Opinions?

Reply:
Yeah, thought about using `__ASSERT` instead of normal `if`s, but `CHECKIF()` is probably the best compromise. I am always a bit wary of adding too many runtime checks even for things which are quite obvious errors and could thus be detected during debugging (e.g. in debug-build-only checks).

We are using CAN also on low-cost STM32F0 series MCUs, where a small-footprint driver implementation is really important.

Reply:
That's fair. I've moved this PR back into draft state until I have reimplemented it using `CHECKIF()`.

Reply:
I have updated this PR to use `CHECKIF()` for the arguments validation, allowing users to disable these run-time checks.

Reply:
This was a good idea.